src/app/Plans/components/Wizard/SelectVMsForm.tsx

```diff
@@ -294,6 +308,11 @@ const SelectVMsForm: React.FunctionComponent<ISelectVMsFormProps> = ({
     }
   });
 
+  const toggleCollapseAll = (collapse: boolean) => {
```

Comment:
@seanforyou23 there is some out of sync state here since the toggle state is separate from the expanded/collapsed state of the rows. For example, if I click the toggle to expand all the rows, but then I collapse an individual row or all the rows by hand, the toggle is still showing in the expanded position.

IMO we don't need the `expandCollapseToggle` state, and we can instead use the `areAllSelected` (renamed to `areAllExpanded`) property from `useSelectionState` to determine which toggle to show. i.e. if all rows are expanded, show it in the AngleDownIcon mode, otherwise the AngleRightIcon mode. This will make it behave like the indeterminate select-all checkbox: if there are any unselected/unexpanded rows, clicking it selects/expands all, and the deselect/collapse-all action is only available if all are selected/expanded.

Reply:
Ah ok, I'm starting to see the value in useSelectionState now. Very cool!